src/launchoptions.rs

```diff
@@ -0,0 +1,211 @@
+pub const USERCON: &str = "-usercon";
+pub const G15: &str = "-g15";
+pub fn get_required_args_vec() -> Vec<String> {
+    let args: &[&str; 5] = &[TF2_GAME_ID, CONDEBUG, CONCLEARLOG, USERCON, G15];
+    let mut buf: Vec<String> = Vec::new();
+    args.iter().for_each(|str| buf.insert(0, str.to_string()));
+    buf
+}
+
+pub struct LaunchOptionsV2 {
+    local_config: PathBuf,
+    target_app: String,
+    launch_args_regex: Regex,
+    app_data: Option<String>,
+    new_app_data: Option<String>,
+    required_options: Vec<String>,
+}
+
+/// TODO: Implement a 'write_new_app_data_to_file' fn to write the updated launch options back to the
+/// file specified by 'local_config' struct element.
+/// TODO: Add more tracing to this to be nicer to the user.
+impl LaunchOptionsV2 {
+    pub fn new(user: SteamID, target_app: String) -> Result<LaunchOptionsV2, anyhow::Error> {
+        let span = tracing::span!(Level::INFO, "LaunchOptions");
+        let _enter = span.enter();
+
+        let required_opts = get_required_args_vec();
+        let config_path: PathBuf = locate_steam_launch_configs(user)
+            .context("Failed to find localconfig.vdf in Steam dir.")?;
+        let mut data: Vec<u8> = Vec::new();
+        tracing::debug!("Found Steam local app config.");
+
+        if let Ok(mut f) = File::open(&config_path) {
+            f.read_to_end(&mut data)
+                .context(format!("Failed to read {:?}", config_path))?;
+        }
+        let binding = &String::from_utf8_lossy(&data);
+
+        let apps_regex_raw = r#"\t{4}"apps"([\s\S]+)\t{5}}"#;
+        let apps_regex =
+            Regex::new(apps_regex_raw).context("Regex construction failed, good work.")?;
+
+        let caps = apps_regex
+            .captures(binding)
+            .context("No capture groups found -  no apps list present in localconfig.vdf.")?;
+
+        let mut matched_app_block: Option<String> = None;
+        let open_match = format!("\t\t\t\t\t\"{}\"", target_app);
+
+        // caps should be an iter of one element.
+        caps.iter().for_each(|mat_opt: Option<Match<'_>>| {
+            if let Some(mat) = mat_opt {
+                let latter_portion = mat
+                    .as_str()
+                    .find(&open_match)
+                    .context(
+                        "Could not find specified app in localconfig.vdf for the current user.",
+                    )
+                    .unwrap();
+                let latter = mat.as_str().substring(latter_portion, mat.end());
+                let first_app_close = latter
+                    .find("\n\t\t\t\t\t}")
+                    .context("Failed to find object closing statement for the matched app block.")
+                    .unwrap();
+                let app_match = latter.substring(0, first_app_close);
+
+                if matched_app_block.is_none() {
+                    matched_app_block = Some(app_match.to_string());
+                }
+            }
+        });
```

Comment:
If you know `caps` only has one capture group (which you do because there's only 1 capture group in the Regex), you can just index into it (0 always being the entire match, so capture groups start from 1).
```rust
let mat = &caps[1];
let latter_portion = mat.find(&open_match)...;
...
```